request/middleware.py

```diff
@@ -11,8 +12,33 @@
 logger = logging.getLogger('request.security.middleware')
 
 
-class RequestMiddleware(MiddlewareMixin):
-    def process_response(self, request, response):
+class RequestMiddleware:
+
+    def __init__(self, get_response):
+        self.get_response = get_response
+
+    def __call__(self, request):
+        start_time = time.time()
+        response = self.get_response(request)
+        end_time = time.time()
+        response_time = end_time - start_time
+        self.create_request_instance(request, response, response_time)
+        return response
+
+    def create_request_instance(self, request, response, response_time=None):
+        """
+        Method is repsonsible for creating a request.Request instance.  It also 
+        provides a hook for capturing the generated instance for anyone that may want to 
+        extend this middleware
+
+        Args:
+            request (django.http.HttpRequest): The django HttpRequest instance
+            response (django.http.HttpResponse): The django HttpResponse instance
+            response_time (float, optional): The request/response cycle time. Defaults to None.
+
+        Returns:
+            request.Request: The request.Request instance that was created
```

Comment:
This does not appear to be fully true, under validation handling None may be returned.

Reply:
How would you recommend changing it?

Reply:
Also, I just caught something else.  The "ifs" should also be returning None not the response.